runtime/src/bank.rs

```diff
@@ -1293,6 +1293,33 @@ impl Bank {
         self.epoch_schedule.get_slots_in_epoch(prev_epoch) as f64 / self.slots_per_year
     }
 
+    // Calculates the starting-slot for inflation from the activation slot.
+    // This method assumes that `pico_inflation` will be enabled before `full_inflation`, giving
+    // precedence to the latter. However, since `pico_inflation` is fixed-rate Inflation, should
+    // `pico_inflation` be enabled 2nd, the incorrect start slot provided here should have no
+    // effect on the inflation calculation.
+    fn get_inflation_start_slot(&self) -> Slot {
+        self.feature_set
+            .activated_slot(&feature_set::full_inflation::id())
+            .unwrap_or_else(|| {
+                self.feature_set
+                    .activated_slot(&feature_set::pico_inflation::id())
+                    .unwrap_or(0)
+            })
+    }
+
+    fn get_inflation_num_slots(&self, prev_epoch: Epoch) -> u64 {
+        let inflation_activation_slot = self.get_inflation_start_slot();
+        // Normalize inflation_start to the last slot of the previous epoch to ensure that
+        // slot_in_year includes the complete epoch.
+        let inflation_start_slot = self.epoch_schedule.get_last_slot_in_epoch(
+            self.epoch_schedule
+                .get_epoch(inflation_activation_slot)
+                .saturating_sub(1),
+        );
+        self.epoch_schedule.get_last_slot_in_epoch(prev_epoch) - inflation_start_slot
```

Comment:
here is my paper simulation:

Say, we activated at 50112000 `Epoch: 116, Slots: 0/432000, [50112000..50544000)`.

as a ultimate goal, for the first inflation reward, we should reward for 2 days (= 432000 num_slots) worth duration.

to create the result, we can choose from...:

```
self.get_inflation_start_slot():50112000
activation_epoch: 116,
self.epoch_schedule.get_first_slot_in_epoch(activation_epoch.saturating_sub(1)): 49680000
self.epoch(): 116
prev_epoch: 115
self.epoch_schedule.get_last_slot_in_epoch(prev_epoch): 50111999
```

then finally this formulate returns `431999`:

```
let inflation_genesis_slot = self.epoch_schedule.get_first_slot_in_epoch(activation_epoch.saturating_sub(1)); // 49680000
let inflation_current_slot = self.epoch_schedule.get_last_slot_in_epoch(prev_epoch); // 50111999
let elapsed_slots = inflation_current_slot - inflation_genesis_slot; // => 431999
```

Then consider, next epoch `Epoch: 117, Slots: 0/432000, [50544000..50976000)`

```
self.get_inflation_start_slot():50112000 // should be same
activation_epoch: 116, // should be same
self.epoch_schedule.get_first_slot_in_epoch(activation_epoch.saturating_sub(1)): 49680000 // should be same
self.epoch(): 117
prev_epoch: 116
self.epoch_schedule.get_last_slot_in_epoch(prev_epoch): 50543999
```

then finally this formulate returns `863999`:

```
let inflation_genesis_slot = self.epoch_schedule.get_first_slot_in_epoch(activation_epoch.saturating_sub(1)); // 49680000
let inflation_current_slot = self.epoch_schedule.get_last_slot_in_epoch(prev_epoch); // 50543999
let elapsed_slots = inflation_current_slot - inflation_genesis_slot; // => 863999
```

and so on...

Reply:
the basic idea is that we must start to the `taper` ticking down at the start of the epoch for which inflation is rewarded. so we should kind of need to shift an epoch for it.

Reply:
btw, this adjustment wasn't needed if we're enabling inflation since genesis...
Because, we're starting in the middle and we want to reward at the activation_epoch for the duration `prev_epoch`, we have to be careful for the handling of these, I think...

Reply:
This PR now aligns the taper with the start of rewards accrual.
However, I think that we don't really want to reward at the activation_epoch -- the activation should kick off accruals, and the reward should happen at the next epoch boundary.